src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/CourseActivityRepository.kt

```diff
@@ -26,16 +26,13 @@ interface CourseActivityRepository : JpaRepository<CourseActivity, Long> {
       (
        select distinct(opp.courseActivity.courseActivityId) 
        from OffenderProgramProfile opp
-       join OffenderBooking ob on opp.offenderBooking = ob
        where opp.prison.id = :prisonId 
-       and opp.programStatus.code = 'ALLOC'
-       and (opp.endDate is null or opp.endDate > current_date)
-       and ob.active = true
-       and ob.location.id = :prisonId
+       and opp.startDate is not null
+       and (opp.endDate is null or opp.endDate >= :oldestAllocation)
       )   
```

Comment:
This is the main change. We used to only migrate activities with active allocations for prisoners currently in the prison. This excluded some intermittent activities that only have active allocations infrequently (e.g. induction).

We now migrate activities that have had an allocation in the last 6 months.